ServiceNow custom connector source code

vasavib opened the pull request:
---
Please check the following conditions for your PR.

- [x] `apiDefinition.swagger.json` is validated using `paconn validate` command.
- [x] `apiProperties.json` has a valid brand color. Invalid brand colors are `#007ee5` and `#ffffff`.

vasavib commented:
@schabungbam - here is the Pull Request I created for the ServiceNow custom connector

schabungbam commented on custom-connectors/ServiceNow/Readme.md:
```diff
@@ -0,0 +1,80 @@
+## ServiceNow [Sample] Connector
+This is a sample connector provided here with the intent to illustrate basic create, read, update & delete operations on ServiceNow Incident records, support for OAuth authentication and attachments
+
+Public connector: https://docs.microsoft.com/en-us/connectors/service-now/
```
Make this into bullets so that it shows up on different lines

vasavib replied:
Made this change

schabungbam commented on custom-connectors/ServiceNow/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,3130 @@
+    "description": "A sample connector for ServiceNow with basic CRUD for Incidents, OAuth and attachment support",
+    "contact":{
+      "name": "https://www.microsoft.com/",
+      "url": "https://support.microsoft.com/",
```
I dont think Microsoft Support will handle this. Lets remove this.

vasavib replied:
have removed Microsoft contact info.

schabungbam commented on custom-connectors/ServiceNow/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,3130 @@
+
+    }
+  },
+  "host": "dev91547.service-now.com",
```
Do you want to replace "dev01547" with "<<instance>>"?

schabungbam commented on custom-connectors/ServiceNow/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,3130 @@
+      "type": "oauth2",
+      "flow": "accessCode",
+      "authorizationUrl": "https://dev91547.service-now.com/oauth_auth.do",
+      "tokenUrl": "https://dev91547.service-now.com/oauth_token.do",
```
Here as well.

schabungbam replied:
@ShyamSunder90 , @mamurshe  - looks like we have duplicate entries for security configuration. It seems the UX will stamp the value in the swagger file as well, while the CLI tool generates the api properties. Ideally, we would only require at one place.

schabungbam commented on custom-connectors/ServiceNow/Readme.md:
```diff
@@ -0,0 +1,80 @@
+1. Go to your ServiceNow instance, copy the instance details and update in the next step. 
+2. Replace `instance` in 
+    - apiProperties.json file - under authorizationURL, tokenURL and refreshURL
+    - apiDefinitions.swagger.json file - for host, authorizationURL and tokenURL.  
```
see my comments on the other file regarding 'instance' in the swagger.json file

schabungbam commented on custom-connectors/ServiceNow/Readme.md:
```diff
@@ -0,0 +1,80 @@
+* Deploy the custom connector from GitHub to an environment using the above steps.
+* Create a new app using the sample .msapp file.
+* Edit the app and add a connection to the custom connector from your app. 
+
```
Can we add a couple of sample expressions here when used from an app?

For example:
Listing incidents:
MyGallery1.Items = ServiceNowCustomSample.ListIncidents().results

Also, does the app have example on how to use attachments? It would be good to add that in a next iteration.

vasavib replied:
I have added sample code to ReadMe. I have a screen to show how upload attachment works.

schabungbam reviewed (commented):
The image for the flow doesn't seem to be used.
Overall, looks good with a few comments.
Thanks for submitting the PR.

vasavib commented:
@schabungbam  -  I have updated per your comments. Seems like there is a build validation failure. I thought it was not accepting the <> and changed it to 'YourInstance'. Let me know if I should undo my last commit with that change.

vasavib commented:
I was unsure of the flow and had removed it previously. I made few changes and uploaded the flow package, and added to ReadMe section.

schabungbam reviewed (dismissed):
Looks good to me.

adrwh commented:
Hi, when i import the Connector, enter my instance details, and save, it whinges that i need to enter a value for "scope", however you have not provided any.  I can actually test the connector and it works fine, but i cannot save it without a scope value?
